addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1737,7 +1738,26 @@ export class Wysiwyg extends Component {
                     params.node.setAttribute(attribute.nodeName, attribute.nodeValue);
                 }
             } else {
-                params.node.replaceWith(element);
+                if (this.options.enableTranslation) {
+                    const attributesToKeep = ["data-resize-width", "title", "alt", "data-oe-translation-state"];
+                    // We don't replace the node with the element because there
+                    // are mutation observers needed for the translation system.
+                    for (const attribute of element.attributes) {
+                        if (!attributesToKeep.includes(attribute.name)) {
+                            params.node.setAttribute(attribute.name, attribute.value);
+                        }
+                    }
+                    for (const attribute of params.node.attributes) {
+                        if (!attributesToKeep.includes(attribute.name)
+                            && !element.hasAttribute(attribute.name)
+                        ) {
+                            params.node.removeAttribute(attribute.name);
+                        }
+                    }
+                    element = params.node;
+                } else {
+                    params.node.replaceWith(element);
+                }
```

Comment:
This one should still be safe. We replace by an element we created our self via the media dialog.